### DISCUSSION TITLE
Collision Detection

### DISCUSSION THREAD

**Discussion by mwisnefsky** [genaray/Arch#200] [Q&A]
Hi - noob here who likes playing around with this kind of stuff.  I'm trying to get a collision detection system working, but am unsure how to handle it.  I need to iterate through every pair of Entities that has a Collider component, but am not sure how to use the Arch framework to do this.

First thought was World.Query inside a World.Query but I keep getting an index out of bound error (I think the worldID is out bounds at some point).  Is there a better way to do this?

**Comment by demian-moon** [comment]
Collision Detection은 일반적으로 QuadTree등을 통해서 합니다. 
단순히 World에 생성된 Collider 컴포넌트를 가진 Entitty들을 모두 Collision 검사를 하는건 매우 비효율적이기 때문입니다.
